stable_baselines3/common/evaluation.py

```diff
@@ -75,15 +76,21 @@ def evaluate_policy(
     episode_rewards = []
     episode_lengths = []
 
-    episode_counts = np.zeros(n_envs, dtype="int")
+    observations = env.reset()
+    if isinstance(observations, dict):
+        device = next(iter(observations.values())).device
```

Comment:
I'm confused why we cast to an iter and then call `next` here, and use the first element below in the `elif`.

Like, it seems simpler to have this line as:
```
device = observations.values[0].device
```

unless I'm missing something.

Reply:
I would like to write the code you wrote, but it doesn't work. `observations` is a `dict`. Here's trying what you said:

```py
>>> {'a': 2}.values[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'builtin_function_or_method' object is not subscriptable
>>> {'a': 2}.values()[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_values' object is not subscriptable
```

Reply:
We could do `list({'a': 2}.values())[0]` but at this point we might as well do `next(iter(...))`.

Reply:
Ah gotcha. Thanks! `next(iter(...))` seems fine then.

I think I don't have write access to this project so I can't resolve this comment, but consider it resolved.